src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -22,15 +23,45 @@ public static class ServiceCollectionExtensions
 {
     public static IServiceCollection AddFdc3DesktopAgent(
         this IServiceCollection serviceCollection,
-        Action<Fdc3DesktopAgentConfig>? builderAction = null)
+        Action<Fdc3DesktopAgentBuilder>? builderAction = null)
     {
-        var config = new Fdc3DesktopAgentConfig();
+        var config = new Fdc3DesktopAgentBuilder(serviceCollection);
+
         if (builderAction != null)
         {
             builderAction(config);
         }
+
         serviceCollection.AddSingleton(config);
         serviceCollection.AddSingleton<IHostedService, Fdc3DesktopAgent>();
         return serviceCollection;
     }
+
+    /// <summary>
+    /// Checks the configuration, if that contains Fdc3Options part, where the user could set the EnableFdc3 tag to true.
+    /// If that tag value is true, it will add the DesktopAgent service to the ServiceCollection, with the given Fdc3Options, that was set in the configuration.
+    /// </summary>
+    /// <param name="serviceCollection"></param>
+    /// <param name="configuration"></param>
+    /// <param name="builderAction"></param>
+    /// <returns></returns>
+    public static IServiceCollection InjectFdc3BackendServiceIfEnabledFromConfig(
+        this IServiceCollection serviceCollection,
+        IConfiguration configuration,
+        Action<Fdc3DesktopAgentBuilder>? builderAction = null)
+    {
+        var fdc3Options = new Fdc3Options();
+
+        var fdc3Configuration = configuration.GetSection(Fdc3Options.Fdc3OptionsName);
+
+        fdc3Configuration.Bind(fdc3Options);
```

Comment:
Nitpick: would it be more convenient to simply use ConfigurationBinder.Get<T>?

Reply:
Sure, I will replace it, but the only reason why I haven't used it, because right now the whole configuration is passed to the method, which will look for the `Fdc3Options` IConfigurationSection. And with this solution (with the `ConfigurationBinder`) the user only could pass that section, otherwise - if the user will pass the whole config it will return the default value. I added a descript for this parameter, now that i replaced with `ConfigurationBinder` solution :)

Reply:
I mean you can use `configuration.GetSection(...).Get<T>()`